dev-utils/docker-compose.yml

```diff
@@ -74,46 +89,71 @@ services:
       test: ["CMD", "curl", "--write-out", "'HTTP %{http_code}'", "--silent", "--output", "/dev/null", "http://kibana:5601/"]
       retries: 10
       interval: 10s
+    mem_limit: 2000m
+#    cpus: 0.5
     depends_on:
       elasticsearch:
         condition: service_healthy
     labels:
       - "co.elatic.apm.stack-version=${STACK_VERSION}"
+    networks:
+      apm:
+        aliases:
+          - kibana
 
-  apm-server-dev:
-    build:
-      context: ./apm-server-dev
-      args:
-        APM_SERVER_REPO: ${APM_SERVER_REPO}
-        APM_SERVER_BRANCH: ${APM_SERVER_BRANCH}
-    ports:
-      - "127.0.0.1:${APM_SERVER_PORT:-8200}:8200"
+  wait-for-apm-server:
+    container_name: wait-for-apm-server
+    image: alpine
     command: >
-      ./apm-server -e
-        -E apm-server.rum.enabled=true
-        -E apm-server.rum.rate_limit=100000
-        -E apm-server.host=0.0.0.0:8200
-        -E apm-server.read_timeout=1m
-        -E apm-server.shutdown_timeout=2m
-        -E apm-server.write_timeout=1m
-        -E setup.template.settings.index.number_of_replicas=0
-        -E output.elasticsearch.hosts=http://elasticsearch:9200
-    cap_drop:
-      - ALL
-    cap_add:
-      - CHOWN
-      - DAC_OVERRIDE
-      - SETGID
-      - SETUID
-    logging:
-      driver: 'json-file'
-      options:
-          max-size: '2m'
-          max-file: '5'
-    depends_on:
-      elasticsearch:
-        condition: service_healthy
+      /bin/sh -c exit 0
     healthcheck:
-      test: ["CMD", "curl", "--write-out", "'HTTP %{http_code}'", "--silent", "--output", "/dev/null", "http://apm-server:8200/"]
+      test: ["CMD", "/bin/sh", "-c", "exit 0"]
       retries: 10
-      interval: 10s
\ No newline at end of file
+      interval: 10s
+    depends_on:
+      apm-server:
+        condition: service_healthy
+    networks:
+      apm:
+        aliases:
+          - wait-for-apm-server
+
+  node-puppeteer:
+    container_name: node-puppeteer
+    image: docker.elastic.co/observability-ci/node-puppeteer:8
+    environment:
+      - HOME=/app
+      - SCOPE=${SCOPE}
+      - STACK_VERSION=${STACK_VERSION}
+      - APM_SERVER_URL=${APM_SERVER_URL}
+      - BUILD_NUMBER=${BUILD_NUMBER}
+      - BRANCH_NAME=${BRANCH_NAME}
+      - JENKINS_URL=${JENKINS_URL}
+      - SAUCE_USERNAME=${SAUCE_USERNAME}
+      - SAUCE_ACCESS_KEY=${SAUCE_ACCESS_KEY}
+      - MODE=${MODE}
+    command: >
+      /bin/bash -c "set -x
+        id
+        node --version
+        npm --version
+        npx --version
+        cd /app
+        export
+        yarn install
```

Comment:
it's better to use same package managers everywhere. any reason why `npm install` is not used here?

Reply:
I've had problems with old node Docker containers and `npm install`, I don't remember if here it was the case. It is easy to check, and if there is no error we could change it.

Reply:
confirmed we can change it to `npm install`